wordpress-mobile/AztecEditor-iOS
# FormatBar: Update list and heading pickers

This PR improves the appearance and behaviour of the list and heading pickers in the format bar.

* We now present options for Lists and well as Headers
* Both are based on their respecting formatting types, with some extensions to provide names and icons.
* Menus now contain icons, and reflect the state of the currently selected format.
* Some improvements to the behaviour when switching between lists:
  * If you tap an item (e.g. Header), and then tap it again, the menu will be hidden.
  * If you tap an item (e.g. Header), and then another item that shows a picker (e.g. List), the picker will switch to the other picker (previously it would be hidden).

<img width="375" alt="screen shot 2017-06-15 at 14 06 06" src="https://user-images.githubusercontent.com/4780/27182649-db6d8744-51d3-11e7-819d-ebcffc7efb05.png">

<img width="375" alt="screen shot 2017-06-15 at 14 06 18" src="https://user-images.githubusercontent.com/4780/27182648-db290556-51d3-11e7-8664-054e937c1462.png">

To test:

* Check the interactions described above
* Check switching between heading types, and list types
* Test hiding and then re-showing the various menus

Needs review: @jleandroperez
